### ISSUE TITLE
freerdp-shadow-cli : not working on Fedora 25

### ISSUE THREAD

**Issue by Minastori** [FreeRDP/FreeRDP#3895]
Hi,

On a machine i set up freerdp server, and on an other i run a rdp client. I can connect to the rdp server, but i get black screen, and on the rdp server, i have repeating error like this : 
`[17:16:12:555] [1897:1899] [ERROR][com.freerdp.server.shadow.x11] - X11 error: BadMatch (invalid parameter attributes) Error code: 8, request code: 49, minor code: 0
[17:16:12:618] [1897:1899] [ERROR][com.freerdp.server.shadow.x11] - X11 error: BadMatch (invalid parameter attributes) Error code: 8, request code: 49, minor code: 0
[17:16:13:679] [1897:1899] [ERROR][com.freerdp.server.shadow.x11] - X11 error: BadMatch (invalid parameter attributes) Error code: 8, request code: 49, minor code: 0
[17:16:13:742] [1897:1899] [ERROR][com.freerdp.server.shadow.x11] - X11 error: BadMatch (invalid parameter attributes) Error code: 8, request code: 49, minor code: 0
[17:16:13:803] [1897:1899] [ERROR][com.freerdp.server.shadow.x11] - X11 error: BadMatch (invalid parameter attributes) Error code: 8, request code: 49, minor code: 0
[17:16:13:866] [1897:1899] [ERROR][com.freerdp.server.shadow.x11] - X11 error: BadMatch (invalid parameter attributes) Error code: 8, request code: 49, minor code: 0
[17:16:13:927] [1897:1899] [ERROR][com.freerdp.server.shadow.x11] - X11 error: BadMatch (invalid parameter attributes) Error code: 8, request code: 49, minor code: 0
[17:16:13:989] [1897:1899] [ERROR][com.freerdp.server.shadow.x11] - X11 error: BadMatch (invalid parameter attributes) Error code: 8, request code: 49, minor code: 0
[17:16:13:052] [1897:1899] [ERROR][com.freerdp.server.shadow.x11] - X11 error: BadMatch (invalid parameter attributes) Error code: 8, request code: 49, minor code: 0`
I try any client and any configuration i have the same problem. I precise, obviously on Fedora 25 i am running on Wayland.

**Comment by hardening** [comment]
If you're running under Fedora 25 then freerdp-shadow is used with the X11 adaptation for wayland, this may be the cause (not sure XTest has been ported).

**Comment by Minastori** [comment]
Effectively, freerdp server must run over xwayland. Is there any possibility to run correctly on Fedora 25 ?

**Comment by realjiangms** [comment]
This link might helps:

https://ask.fedoraproject.org/en/question/96134/how-can-i-set-xorg-as-a-default-on-fedora-25/

**Comment by akallabeth** [comment]
shadow server is X11 only, for wayland use the native RDP backend (we can´t grab the screen like X11 for security reasons)

**Comment by user-72632** [comment]
@hardening Hi, I know this is off topic here but I was interested in how the use of freerdp in the wayland rdp backend is progressing. Lastly, using Debian 10 and wayland, how do you use the freerdp clients to connect to the rdp backend in wayland. I.e., what command line option are needed in freerdp and what setup of wayland on the Debian side are required. I am sure there is security issues and certs to address for doing this.

Sorry again if this is way off topic, just interested and know that you have been integral in its implementation. Thanks again for any help in this setup.

**Comment by mdavidjohnson** [comment]
I'm experiencing an almost identical difficulty between Debian and Windows.

I have Debian 12 (Bookworm) running on my 10.0.0.93 server at home. I did "sudo apt-get install freerdp2-x11" and "sudo apt-get install freerdp2-shadow-x11".

Then I started the RDP server with "sudo freerdp-shadow-cli /bind-address:10.0.0.93 /sec:rdp /may-view /may-interact -auth". It responded "Listening on [10.0.0.93]:3389".

Then I went to my Windows 11 desktop (in the same room, at 10.0.0.100), started RDP and connected to "10.0.0.93:3389". It connected right away, but the screen was (and remained) entirely black.

Back on my Debian machine, the terminal was continuously reporting "[ERROR][com.freerdp.server.shadow.x11] - X11 error: BadMatch (invalid parameter attributes) Error code: 8, request code: 49, minor code: 0" over and over.
